### PR TITLE
Java: Add `SUNIONSTORE` command.

### PR DESCRIPTION
**Pull request by Yury-Fridlyand**
https://redis.io/docs/latest/commands/sunionstore/

### CODE REVIEW

File: java/client/src/main/java/glide/api/commands/SetBaseCommands.java [hunk 1]
```diff
@@ -95,4 +95,19 @@ public interface SetBaseCommands {
      * }</pre>
      */
     CompletableFuture<Boolean> sismember(String key, String member);
+
+    /**
+     * Stores the members of the set resulting from the union of all the given sets into new set.
```
Comment (SanHalacogluImproving) [inline java/client/src/main/java/glide/api/commands/SetBaseCommands.java:100]: Stores the members of the union of all given sets specified by `keys` into a new set at `destination`.

File: java/client/src/main/java/glide/api/commands/SetBaseCommands.java [hunk 2]
```diff
@@ -95,4 +95,19 @@ public interface SetBaseCommands {
...
+     *
+     * @see <a href="https://redis.io/commands/sunionstore/">redis.io</a> for details.
+     * @param destination The key of the destination set where the merged sets will be stored.
```
Comment (SanHalacogluImproving) [inline java/client/src/main/java/glide/api/commands/SetBaseCommands.java:103]: The key of the destination set. should be enough

File: java/client/src/main/java/glide/api/commands/SetBaseCommands.java [hunk 3]
```diff
@@ -95,4 +95,19 @@ public interface SetBaseCommands {
...
+     * @param keys The key from which to retrieve the set members.
```
Comment (SanHalacogluImproving) [inline java/client/src/main/java/glide/api/commands/SetBaseCommands.java:104]: The keys ...

File: java/client/src/main/java/glide/api/commands/SetBaseCommands.java [hunk 4]
```diff
@@ -95,4 +95,19 @@ public interface SetBaseCommands {
...
+     * @return The number of elements in the resulting set.
+     * @example
+     *     <pre>{@code
+     * long length = client.sunionstore("mySet", new String[] { "set1", "set2" }).get();
```
Comment (SanHalacogluImproving) [inline java/client/src/main/java/glide/api/commands/SetBaseCommands.java:108]: Long

File: java/client/src/main/java/glide/api/commands/SetBaseCommands.java [hunk 5]
```diff
@@ -95,4 +95,19 @@ public interface SetBaseCommands {
...
+     * assert length == 5;
```
Comment (SanHalacogluImproving) [inline java/client/src/main/java/glide/api/commands/SetBaseCommands.java:109]: 5L

File: java/client/src/main/java/glide/api/models/BaseTransaction.java [hunk 6]
```diff
@@ -878,6 +879,20 @@ public T scard(@NonNull String key) {
         return getThis();
     }
 
+    /**
+     * Stores the members of the set resulting from the union of all the given sets into new set.
```
Comment (SanHalacogluImproving) [inline java/client/src/main/java/glide/api/models/BaseTransaction.java:883]: Comments as above.

File: java/integTest/src/test/java/glide/SharedCommandTests.java [hunk 7]
```diff
@@ -843,6 +843,49 @@ public void sismember(BaseClient client) {
...
+        assertEquals(3, client.sadd(key1, new String[] {"a", "b", "c"}).get());
+        assertEquals(3, client.sadd(key2, new String[] {"c", "d", "e"}).get());
+        assertEquals(3, client.sadd(key4, new String[] {"e", "f", "g"}).get());
+        // create new
+        assertEquals(5, client.sunionstore(key3, new String[] {key1, key2}).get());
+        assertEquals(Set.of("a", "b", "c", "d", "e"), client.smembers(key3).get());
+        // overwrite existing set
+        assertEquals(5, client.sunionstore(key2, new String[] {key3, key2}).get());
+        assertEquals(Set.of("a", "b", "c", "d", "e"), client.smembers(key2).get());
+        // overwrite source
```
Comment (SanHalacogluImproving) [inline java/integTest/src/test/java/glide/SharedCommandTests.java:865]: Don't believe you need these test where you overwrite keys. It's not bad to have them I agree but doesn't seem like they had tests like these in Node and PY. They have more concise IT tests.

Comment (SanHalacogluImproving) [reply java/integTest/src/test/java/glide/SharedCommandTests.java:865]: @acarbonetto Opinion?

File: java/integTest/src/test/java/glide/SharedCommandTests.java [hunk 8]
```diff
@@ -843,6 +843,49 @@ public void sismember(BaseClient client) {
...
+        assertEquals(6, client.sunionstore(key1, new String[] {key1, key4}).get());
+        assertEquals(Set.of("a", "b", "c", "e", "f", "g"), client.smembers(key1).get());
+        // overwrite source
```
Comment (jonathanl-bq) [inline java/integTest/src/test/java/glide/SharedCommandTests.java:868]: nit: newlines above all comments to stay consistent.

**Review by jonathanl-bq — APPROVED** [review]
Conditional approval if San's comments are addressed.

File: java/client/src/main/java/glide/api/commands/SetBaseCommands.java (hunk 5 above)
Comment (Yury-Fridlyand) [reply java/client/src/main/java/glide/api/commands/SetBaseCommands.java:109]: Both are valid if we are within int range

**Review by SanHalacogluImproving — APPROVED** [review]

**Pull request merged**